### ISSUE TITLE
Windows 7: Title in Electron window not (always) updated

### ISSUE THREAD

**Issue by ogo-adp** [electron/electron#10897]
* Electron version: 1.8.0, 1.8.2 (and probably others)
* Operating system: Windows 7

### Expected behavior

Text in main window title bar should update when changing document.title

### Actual behavior

Title bar does update correctly when Aero is activated on Windows 7.
Title bar does never update when using Electron app in remote desktop or when Aero is not activated. You can get the title to update by moving the window out of screen and back again, so it is only a redraw problem probably specific to Windows port.

### How to reproduce

Launch any Electron app that modifies title (go to www.google.com and nagivate to some url)

**Comment by felixrieseberg** [comment]
We've seen this behavior too, but it seems to be dependent _on the specific machine_ the app runs on, meaning that this bug has to be a bit more subtle.

**Comment by ogo-adp** [comment]
We see this bug on many machines, in fact every Windows 7 machine that has Aero disables (ie classic look).

**Comment by mtsh1000** [comment]
I confirm that. I normally use classic look and this bug is present (the window title updates only after resize or receiving focus from another window), but if I turn on aero its fixed.

**Comment by bpasero** [comment]
VS Code is also seeing this from many users but without luck to reproduce. Would love to see this fixed 👍

**Comment by ogo-adp** [comment]
@bpasero I think you can reproduce it by disabling aero on connecting to a win7 machine using remote desktop. The bug has reproduced every time and dozens of win7 machines by doing this.

**Comment by ogo-adp** [comment]
I looked at the code and could not understand how the window title can be updated ...

There is not call to SetWindowTitle() API, and NativeWindowViews::SetTitle() calls window_->UpdateWindowTitle() which does nothing (empty function).

**Comment by ogo-adp** [comment]
Any news on advance on solving this bug ? @mtsh1000 confirmed that classic look is sufficient to reproduce this behavior. 

Sorry but im y opinion this is a huge bug and the fix should be easy for someone that know well the code base of electron (I did look at it without luck)

**Comment by tariqporter** [comment]
I also have this issue when connecting to remote computers with TeamViewer. Assuming it turns Aero off

**Comment by ogo-adp** [comment]
Bug still present in 2.0.0 b1

Any hope of a fix ?

**Comment by LiPengfei19820619** [comment]
Is there  a plan to fix? thanks.

**Comment by ogo-adp** [comment]
Bug still there is 2.0.0

**Comment by mtsh1000** [comment]
just noticed that the bug is fixed in atom 1.28.1 x64

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v4.2.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons. This issue will be closed 7 days from now if there is no response.

Thanks in advance! Your help is appreciated.

**Comment by bpasero** [comment]
I cannot tell unfortunately.

**Comment by sofianguy** [comment]
Thank you for your issue!

We haven't gotten a response to our questions in our comment above. With only the information that is currently in the issue, we don't have enough information to take action. I'm going to close this but don't hesitate to reach out if you have or find the answers we need, we'll be happy to reopen the issue.
